apps/crn-server/src/data-providers/contentful/event.data-provider.ts

```diff
@@ -0,0 +1,419 @@
+  EventSpeakerUserData,
+  EventUpdateDataObject,
+  FetchEventsOptions,
+  GroupResponse,
+  isEventStatus,
+  ListEventDataObject,
+  WorkingGroupResponse,
+} from '@asap-hub/model';
+import { DateTime } from 'luxon';
+
+import {
+  getContentfulEventMaterial,
+  MeetingMaterial,
+  parseContentfulGraphqlCalendarPartialToDataObject,
+} from '../../entities';
+import { parseCalendarDataObjectToResponse } from '../../controllers/calendars';
+import { waitForUpdated } from '../../utils/wait-for-updated';
+
+export type EventItem = NonNullable<
+  NonNullable<FetchEventsQuery['eventsCollection']>['items'][number]
+>;
+
+export class EventContentfulDataProvider implements EventDataProvider {
+  constructor(
+    private contentfulClient: GraphQLClient,
+    private getRestClient: () => Promise<Environment>,
+  ) {}
+
+  private fetchEventById(id: string) {
+    return this.contentfulClient.request<
+      FetchEventByIdQuery,
+      FetchEventByIdQueryVariables
+    >(FETCH_EVENT_BY_ID, { id });
+  }
+
+  async fetchById(id: string): Promise<EventDataObject | null> {
+    const { events } = await this.fetchEventById(id);
+
+    if (!events) {
+      return null;
+    }
+
+    return parseGraphQLEvent(events);
+  }
+
+  async fetch(options: FetchEventsOptions): Promise<ListEventDataObject> {
+    const {
+      take = 10,
+      skip = 0,
+      before,
+      after,
+      search,
+      sortBy,
+      sortOrder,
+      filter,
+    } = options;
+
+    if (filter?.userId) {
+      const { users } = await this.contentfulClient.request<
+        FetchEventsByUserIdQuery,
+        FetchEventsByUserIdQueryVariables
+      >(FETCH_EVENTS_BY_USER_ID, {
+        limit: take,
+        skip,
+        id: filter.userId,
+      });
+
+      const eventsCollection =
+        users?.linkedFrom?.eventSpeakersCollection?.items[0]?.linkedFrom
+          ?.eventsCollection;
+
+      return getEventDataObject(eventsCollection);
+    }
+
+    if (filter?.externalAuthorId) {
+      const { externalAuthors } = await this.contentfulClient.request<
+        FetchEventsByExternalAuthorIdQuery,
+        FetchEventsByExternalAuthorIdQueryVariables
+      >(FETCH_EVENTS_BY_EXTERNAL_AUTHOR_ID, {
+        limit: take,
+        skip,
+        id: filter.externalAuthorId,
+      });
+
+      const eventsCollection =
+        externalAuthors?.linkedFrom?.eventSpeakersCollection?.items[0]
+          ?.linkedFrom?.eventsCollection;
+
+      return getEventDataObject(eventsCollection);
+    }
+
+    if (filter?.teamId) {
+      const { teams } = await this.contentfulClient.request<
+        FetchEventsByTeamIdQuery,
+        FetchEventsByTeamIdQueryVariables
+      >(FETCH_EVENTS_BY_TEAM_ID, {
+        limit: take,
+        skip,
+        id: filter.teamId,
+      });
+
+      const eventsCollection =
+        teams?.linkedFrom?.eventSpeakersCollection?.items[0]?.linkedFrom
+          ?.eventsCollection;
+
+      return getEventDataObject(eventsCollection);
+    }
+
```

Comment:
maybe add another TODO here somewhere so we know we need to add filtering by interest-groups and working-groups?

Reply:
I've added here https://github.com/yldio/asap-hub/blob/CT-8-events-data-provider/apps/crn-server/src/data-providers/contentful/event.data-provider.ts#L397:L398 :)